src/event-stream/tokens-contract-handler.ts

```diff
@@ -0,0 +1,417 @@
+    }
+  }
+
+  /**helpng method for creating http url */
+  private makeHostedUrl(uri: string): string {
+    if (uri.includes('http')) return uri;
+    const parsedURI = URI.parse(uri);
+    return `${PUBLIC_IPFS}/${parsedURI.host}${parsedURI.path}`;
+  }
+
+  async makeApiCall<Type>(url: string): Promise<Type> {
+    const result = await fetch(url);
+    if (!result.ok) {
+      let msg = '';
+      try {
+        msg = await result.text();
+      } catch (error) {
+        logError(`Error getting text`, error);
+      }
+      throw new Error(`Response ${result.status}: ${result.statusText} fetching ${url} - ${msg}`);
+    }
+    try {
+      const resultString = await result.text();
+      return JSON.parse(resultString) as Type;
+    } catch (error) {
+      logError(`Error reading response from ${url}`, error);
+      throw error;
+    }
+  }
+  /**
+   * fetch metadata from uri
+   */
+  private async getMetadataFromUri<Type>(token_uri: string): Promise<Type> {
+    return await this.makeApiCall(this.makeHostedUrl(token_uri));
+  }
+
+  /**
+   * make readonly contract call
+   */
+
+  private async makeReadOnlyContractCall(
+    functionName: string,
+    functionArgs: ClarityValue[]
+  ): Promise<ClarityValue> {
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: this.address,
+      contractAddress: this.contractAddress,
+      contractName: this.contractName,
+      functionName: functionName,
+      functionArgs: functionArgs,
+      network: this.stacksNetwork,
+    };
+    return await callReadOnlyFunction(txOptions);
+  }
+
+  /**
+   *Store ft metadata to db
+   */
+  private async storeFTMetadata(ft_metadata: DbFungibleTokenMetadata) {
+    try {
+      await this.db.updateFtMetadata(ft_metadata);
+    } catch (error) {
+      logger.error(`Error: error occured while updating FT metadata`);
+    }
+  }
+
+  /**
+   *store NFT Metadata to db
+   */
+  private async storeNFTMetadata(nft_metadata: DbNonFungibleTokenMetadata) {
+    try {
+      await this.db.updateNFtMetadata(nft_metadata);
+    } catch (error) {
+      logger.error(`Error: error occured while updating NFT metadata`);
+    }
+  }
+
+  /**
+   * This method check if the contract is compliance with sip-09 and sip-10
+   * Ref: https://github.com/stacksgov/sips/tree/main/sips
+   */
+  private isCompliant(standardFunction: ClarityAbiFunction[]): boolean {
+    return standardFunction.every(abiFun => this.findFunction(abiFun, this.contractAbi.functions));
+  }
+
+  /**
+   * check if the fun  exist in the function list
+   * @param fun - function to be found
+   * @param functionList - list of functions
+   * @returns - true if function is in the list false otherwise
+   */
+  private findFunction(fun: ClarityAbiFunction, functionList: ClarityAbiFunction[]): boolean {
+    const found = functionList.find(standardFunction => {
+      if (standardFunction.name != fun.name || standardFunction.args.length != fun.args.length)
```

Comment:
Should use strict equality `!==`. We should get the eslint rule fixed for this (in a later PR)